### ISSUE TITLE
[Bug] Bookkeeping of device actors failed upon deletion

### ISSUE THREAD

**Issue by srvaroa** [thingsboard/thingsboard#8686]
**Describe the bug**

TB node logs report having known devices for which it is calculating state updates, however there are no devices in the system. This happens after deleting a full tenant.

**Your Server Environment**

* own setup
  * docker deployment (we used a modified version of the helm chart in thingsboard-ce-k8s)
  * TB 3.5.0

* Connectivity
  * MQTT

**To Reproduce**

We have reproduced this issue consistently in a local environment, with both TB 3.3.4 and later after upgrading to 3.5.0.

We start an environment, with no devices connected.

We run a script that creates 1 tenant, and connects ~200 devices. Each device is a docker image that simulates a real device and connects to the MQTT broker, etc. (this docker image is well tested and in use internally for production environments).

The script runs a bunch of tests, and stops. Then we delete the tenant and we see all devices disappear from the UI (as expected). We have no other tenants or devices in this environment.

![image](https://github.com/thingsboard/thingsboard/assets/346110/e4093ee1-fe45-4245-a4fe-434a1a10360d)

However, we see that tb-node logs continues trying to figure out state for devices, even if they are no longer there:

![image](https://github.com/thingsboard/thingsboard/assets/346110/9b532864-f8b9-4d3e-b8a4-23702d0db138)

We observe that in the other TB-nodes we are running (3 in total), the devices reported are 0. However, the node that still sees devices doesn't seem to realize that they are gone.

**Expected behavior**

After the tenant is deleted, TB nodes should stop checking for state of those devices.

**Comment by ViacheslavKlimov** [comment]
Are these logs about non-existent devices continue to be printed each minute after tenant deletion? Or are they printed only once and further logs show correct info?

**Comment by srvaroa** [comment]
Hi, they are printed continuously

**Comment by mykhailokornieiev** [comment]
Hi. I was unable to reproduce this behavior on the latest version.
As the community has not provided assistance on this matter over an extended period, I am closing this ticket due to inactivity. If it is still relevant, feel free to open a new ticket with a full description of your request and full detailed steps to reproduce it.
